crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -72,18 +72,18 @@ struct SyncServiceInner {
     encryption_sync_service: Arc<EncryptionSyncService>,
     state: SharedObservable<State>,
     encryption_sync_permit: Arc<AsyncMutex<EncryptionSyncPermit>>,
-    /// Scheduler task ensuring proper termination.
+    /// Supervisor task ensuring proper termination.
```

Comment:
I agree it's a better name according to your definitions. _Scheduler_ wasn't “wrong” neither, but I believe _Supervisor_ is more adapted.